pages/api/user/index.ts

```diff
@@ -18,10 +21,29 @@ export default async function handle(
             break;
         }
         case "POST": {
+            // TODO
             break;
         }
         case "PUT": {
-            // TODO:
+            const session = await getSession({ req });
+            const userId = session.id as string;
+            // TODO: add user role to session
+            const updatedUserData = {
+                id: userId,
+                firstName: req.body.firstName,
+                lastName: req.body.lastName,
+                role: req.body.role,
+                roleData: req.body.roleData,
+            } as UserInput;
```

Comment:
instead of typecasting this - you can just define it as a UserInput

`const updatedUserData: UserInput = {...}`